ember-bootstrap/addon/template-registry.ts

```diff
@@ -1,9 +1,36 @@
 import type BsButton from './components/bs-button';
 import type BsCollapse from './components/bs-collapse';
+import type BsModal from './components/bs-modal';
+import type BsModalBody from './components/bs-modal/body';
+import type BsModalDialog from './components/bs-modal/footer';
+import type BsModalFooter from './components/bs-modal/footer';
+import type BsModalHeader from './components/bs-modal/header';
+import type BsModalHeaderClose from './components/bs-modal/header/close';
+import type BsModalHeaderTitle from './components/bs-modal/header/title';
 import type BsSpinner from './components/bs-spinner';
 
-export default interface EmberBootstrapRegistry {
+interface BsModalRegistry {
+  'bs-modal/header/close': typeof BsModalHeaderClose;
+  'BsModal::Header::Close': typeof BsModalHeaderClose;
+  'bs-modal/header/title': typeof BsModalHeaderTitle;
+  'BsModal::Header::Title': typeof BsModalHeaderTitle;
+  'bs-modal/header': typeof BsModalHeader;
+  'BsModal::Header': typeof BsModalHeader;
+  'bs-modal/body': typeof BsModalBody;
+  'BsModal::Body': typeof BsModalBody;
+  'bs-modal/footer': typeof BsModalFooter;
+  'BsModal::Footer': typeof BsModalFooter;
+  'bs-modal/dialog': typeof BsModalDialog;
+  'BsModal::Dialog': typeof BsModalDialog;
+  'bs-modal': typeof BsModal;
+  BsModal: typeof BsModal;
+}
+
+export default interface EmberBootstrapRegistry extends BsModalRegistry {
+  'bs-button': typeof BsButton;
   BsButton: typeof BsButton;
+  'bs-collapse': typeof BsCollapse;
   BsCollapse: typeof BsCollapse;
+  'bs-spinner': typeof BsSpinner;
   BsSpinner: typeof BsSpinner;
 }
```

Comment:
Why do we add all the dasherized versions? Why do we add all the private components yielded such as `BsModall::Header`? I expected that it only adds `BsModal: typeof BsModal;` to the existing list.

This review comment from @simonihmig seems to be touching the same topic in another PR: https://github.com/ember-bootstrap/ember-bootstrap/pull/2164/files#r1868407700

Reply:
You're right. The dasherized versions should not be part of the public API. These need to be moved to the internal types until we can get rid of the component helper usages.